components/dashboard/src/onboarding/StepUserInfo.tsx

```diff
@@ -0,0 +1,113 @@
+import { OnboardingStep } from "./OnboardingStep";
+
+type Props = {
+    user: User;
+    onComplete(user: User): void;
+};
+export const StepUserInfo: FC<Props> = ({ user, onComplete }) => {
+    const updateUser = useUpdateCurrentUserMutation();
+    // attempt to split provided name for default input values
+    const { first, last } = getInitialNameParts(user);
+
+    const [firstName, setFirstName] = useState(first);
+    const [lastName, setLastName] = useState(last);
+    const [emailAddress, setEmailAddress] = useState(User.getPrimaryEmail(user) ?? "");
+
+    const handleSubmit = useCallback(async () => {
+        const additionalData = user.additionalData || {};
+        const profile = additionalData.profile || {};
+
+        const updates = {
+            // we only split these out currently for form collection, but combine in the db
+            fullName: `${firstName} ${lastName}`,
+            additionalData: {
+                ...additionalData,
+                profile: {
+                    ...profile,
+                    emailAddress,
+                    lastUpdatedDetailsNudge: new Date().toISOString(),
+                },
+            },
+        };
+
+        try {
+            const updatedUser = await updateUser.mutateAsync(updates);
+            onComplete(updatedUser);
+        } catch (e) {
+            console.error(e);
+        }
+    }, [emailAddress, firstName, lastName, onComplete, updateUser, user.additionalData]);
+
+    const firstNameError = useOnBlurError("Please enter a value", !!firstName);
+    const lastNameError = useOnBlurError("Please enter a value", !!lastName);
+    const emailError = useOnBlurError("Please enter your email address", !!emailAddress);
+
+    const isValid = [firstNameError, lastNameError, emailError].every((e) => e.isValid);
+
+    return (
+        <OnboardingStep
+            title="Get started with Gitpod"
+            subtitle="Fill in the name and email you want to use to author commits."
+            error={updateUser.isError ? "There was a problem updating your profile" : undefined}
+            isValid={isValid}
+            isLoading={updateUser.isLoading}
+            onSubmit={handleSubmit}
+        >
+            <div className="flex justify-between space-x-2 w-full">
+                <TextInputField
+                    containerClassName="w-1/2"
+                    value={firstName}
+                    label="First name"
+                    error={firstNameError.message}
+                    onBlur={firstNameError.onBlur}
+                    onChange={setFirstName}
+                />
+
+                <TextInputField
+                    containerClassName="w-1/2"
+                    value={lastName}
+                    label="Last name"
+                    error={lastNameError.message}
+                    onBlur={lastNameError.onBlur}
+                    onChange={setLastName}
+                />
```

Comment:
nit: I think some things in here are missing that would be nice to add:
- [`autoComplete`](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete) - to make it easier for users to fill it in
- [`required`](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/required) - make the inputs accessible by not just doing input validation, but signalling to the browser, that these input fields must be filled in

Reply:
Let's add this in a follow-up PR, as it's nothing blocking.

Reply:
Added `required`, good call. `autoComplete` is `"on"` by default in browsers, so don't need to explicitly add specify it unless you want to disable it. 

I did add `type="submit` to the button, but most browsers don't require that for the autocomplete hueristic anymore (but good to be explicit there anyways). Some other conditions browsers use to decide if it should provide autocomplete that include things like, was the input empty initially (if not, it won't offer autocomplete), does the input have a `name` attribute (most browsers don't require this one anymore afaik, but it'd be a good thing for us to add in a followup)